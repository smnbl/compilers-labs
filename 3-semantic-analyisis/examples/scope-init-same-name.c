// RUN-WITH-ARGS: -dump-symbol-table

int foo(int x)
{
    int x = x + 2; // shadows the function parameter, and is initialised with the parameter's value + 2
    return x; // returns the local variable
}
