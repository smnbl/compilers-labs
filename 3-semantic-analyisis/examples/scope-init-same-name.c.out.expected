└── Program <9>
    └── FuncDecl: returnType = 'int', name = 'foo' <8>
        ├── VarDecl: type = 'int', name = 'x' <0>
        └── CompoundStmt <7>
            ├── VarDecl: type = 'int', name = 'x' <4>
            │   └── BinaryOpExpr: op = '+' <3>
            │       ├── VarRefExpr: name = 'x' <1>
            │       └── IntLiteral: value = '2' <2>
            └── ReturnStmt <6>
                └── VarRefExpr: name = 'x' <5>
Symbol table:
1                   0                   
5                   4                   
